CMakeLists.txt

```diff
@@ -77,6 +78,32 @@ if(NOT MSVC)
     endif()
 endif()
 
+# This is copied wholesale from the implementation of LLVM_ENABLE_ASSERTIONS
```

Comment:
does this need some kind of extra copyright statement or something? I'm not *not* going to steal this, it's a nice known good implementation that does exactly what we need

Reply:
Probably, it would be easier to put this in a separate file e.g. `cmake/Assertions.cmake` and add a separate license notice in that file, then include it here.

Reply:
I've done that, hopefully its sufficient

Reply:
Looks good, could also add a permalink to the line in the file where the code came from in the license comment to clarify of source.